src/androidTest/java/de/blau/android/dialogs/BookmarkDialogsTest.java

```diff
@@ -0,0 +1,116 @@
+public class BookmarkDialogsTest {
+
+    Instrumentation instrumentation = null;
+    ArrayList<BookmarksStorage> bookmarksStorages = null;
+    ViewBox     viewBoxtest = null;
+    ActivityMonitor monitor = null;
+    UiDevice        device  = null;
+    Splash          splash  = null;
+    Main            main    = null;
+    Map             map     = null;
+
+    @Rule
+    public ActivityTestRule<Splash> mActivityRule = new ActivityTestRule<>(Splash.class, false, false);
+
+    /**
+     * Pre-test setup
+     */
+    @Before
+    public void setup() {
+        instrumentation = InstrumentationRegistry.getInstrumentation();
+        device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+        monitor = instrumentation.addMonitor(Main.class.getName(), null, false);
+        Intent intent = new Intent(Intent.ACTION_MAIN);
+        splash = mActivityRule.launchActivity(intent);
+        main = (Main) instrumentation.waitForMonitorWithTimeout(monitor, 90000);// NOSONAR wait for main
+        assertNotNull(main);
+        TestUtils.grantPermissons(device);
+        map = main.getMap();
+        TestUtils.dismissStartUpDialogs(device, main);
+        TestUtils.stopEasyEdit(main);
+        bookmarksStorages = new ArrayList<>();
+        try{
+            //India
+            bookmarksStorages.add(new BookmarksStorage("TestLocation0",new ViewBox(68.1766451354,7.96553477623,97.4025614766,35.4940095078)));
+            //Netherlands
+            bookmarksStorages.add(new BookmarksStorage("TestLocation1",new ViewBox(3.31497114423,50.803721015,7.09205325687,53.5104033474)));
+            //Serbia
+            bookmarksStorages.add(new BookmarksStorage("TestLocation2",new ViewBox(18.82982,42.2452243971,22.9860185076,46.1717298447)));
+        }
+        catch (OsmException osmex){
+            osmex.printStackTrace();
+        }
+    }
+
+    /**
+     * Checks if the the viewboxes are same across dialogs
+     */
+    @Test
+    public void AddRemoveTest() {
+        //Add Dialog
+        for(int i = 0;i<3;i++){
+            map.getViewBox().fitToBoundingBox(map,bookmarksStorages.get(i).viewBox);
+            map.invalidate();
+            Assert.assertTrue(TestUtils.clickResource(device, true, device.getCurrentPackageName() + ":id/menu_gps", true));
+            Assert.assertTrue(TestUtils.clickText(device, false, "Add Bookmark..", true, false));
+            UiObject comments = device.findObject(new UiSelector().clickable(true).resourceId(device.getCurrentPackageName() + ":id/text_line_edit"));
+            try {
+                comments.click();
+                comments.setText("TestLocation"+i);
+            } catch (UiObjectNotFoundException e) {
+                Assert.fail(e.getMessage());
+            }
+            TestUtils.clickButton(device, "android:id/button1", true);
+        }
+        //Show Dialog
+        for(int i =0;i<3;i++){
+            TestUtils.clickMenuButton(device, "GPS/GNSS…", false, true);
```

Comment:
While we are no way consistent in this, it is in principle good practice to use the resource ids and get String to get the translated string. Right now the tests will not run in any language than English due to this not being done in many places, but we can work towards this being possible.